DatadogTrace/Sources/TraceConfiguration.swift

```diff
@@ -0,0 +1,151 @@
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+// Export `DDURLSessionDelegate` elements to be available with `import DatadogTrace`:
+@_exported import class DatadogInternal.DatadogURLSessionDelegate
+@_exported import typealias DatadogInternal.DDURLSessionDelegate
+@_exported import protocol DatadogInternal.__URLSessionDelegateProviding
+
+extension Trace {
+    /// Trace feature configuration.
+    public struct Configuration {
+        public typealias EventMapper = (SpanEvent) -> SpanEvent
+
+        /// The sampling rate for spans created with the default tracer.
+        ///
+        /// It must be a number between 0.0 and 100.0, where 0 means no spans will be collected.
+        ///
+        /// Default: `100.0`.
+        public var sampleRate: Float
+
+        /// The `service` value for spans.
+        ///
+        /// If not specified, the SDK default `service` will be used.
+        public var service: String?
+
+        /// Global tags associated with each span created with the default tracer.
+        public var tags: [String: Encodable]?
+
+        /// The configuration for automatic network requests tracing.
+        ///
+        /// Requests tracing requires using the `DDURLSessionDelegate` in `URLSessions` that you want to trace.
+        ///
+        /// Note: Automatic network tracing involves swizzling the `URLSession` methods and will not work
+        /// without using `DDURLSessionDelegate`.
+        ///
+        /// Default: `nil` - which means automatic tracing is not enabled by default.
+        public var urlSessionTracking: URLSessionTracking?
+
+        /// Enables the integration of traces with RUM.
+        ///
+        /// When enabled, all spans will be enriched with the current RUM view information.
+        ///
+        /// Default: `true`.
+        public var bundleWithRUM: Bool
+
+        /// Enriches traces with network connection information.
+        ///
+        /// This includes reachability status, connection type, mobile carrier name, and more, which will be added to every span and span log.
+        ///
+        /// Default: `false`
+        public var sendNetworkInfo: Bool
+
+        /// Custom mapper for span events.
+        ///
+        /// It can be used to modify span events before they are sent. The implementation of the mapper should
+        /// obtain a mutable copy of `SpanEvent`, modify it, and return it. Keep the implementation fast
+        /// and do not make any assumptions on the thread used to run it.
+        ///
+        /// Default: `nil`.
+        public var eventMapper: EventMapper?
+
+        /// Custom server url for sending traces.
+        ///
+        /// Default: `nil`.
+        public var customEndpoint: URL?
+
+        // MARK: - Nested Types
+
+        /// Configuration of automatic network requests tracing.
+        public struct URLSessionTracking {
```

Comment:
I wonder if shouldn't combine this one and `RUM.Configuration.URLSessionTracking` into a single config struct defined in `DatadogInternal` 🤔 

Not a change request but I would like your opinion on this?

Reply:
I was thinking about this 👌, but IMO this wouldn't be correct nor user-friendly.

First, there is a slight but key difference in both:
```swift
// DatadogRUM:
public struct URLSessionTracking {
    public var firstPartyHostsTracing: FirstPartyHostsTracing? // <-- ❗️ optional
    public var resourceAttributesProvider: RUM.ResourceAttributesProvider?
}

// DatadogTrace:
public struct URLSessionTracking {
    public var firstPartyHostsTracing: FirstPartyHostsTracing // <-- ❗️ non-optional
}
```
Not only RUM takes `resourceAttributesProvider`, but also optionality of `firstPartyHostsTracing` is different. This comes from the fact that:
* If `URLSession` is instrumented by RUM, **it tracks all requests** as RUM resource. Additionally it will inject tracing headers to first-party requests, hence we ask for their definition optionaly. 
* If `URLSession` is instrumented by Trace, **it doesn't track all requests** (ref.: #314). It only intercepts first-party requests and injects tracing headers. Hence, it doesn't make sense to have `URLSessionTracking` without definition of hosts.

Next, because of that difference ☝️ , it could be hard to come up with a single definition of `URLSessionTracking` that works for both products and includes product-agnostic explanation in API comment.

Last, this ☝️ is the difference we have today. It is easy to imagine this drifting even more away (e.g. distinguishing the `sampleRate` value in `FirstPartyHostsTracing` enum of both products). Consider the fact that RUM and APM are separate DD products with distinct infrastructure which might translate into separate constraints.

For all that reasons, I think it is safer duplicate this definition than assume wrong abstraction. To help users with adopting this, I made sure that both `URLSessionTracking` and `FirstPartyHostsTracing` APIs stay close to each other in both products.

How does it all sound @maxep ?

Reply:
Also, keep in mind that there is no practical use case of "sharing config definition" by the user. RUM offers a superset of distributed tracing - if you enable automatic RUM resources with `urlSessionTracking`, no real point of enabling APM network tracing as RUM will manage do distributed tracing for you.

Reply:
It all makes sense, thanks for the clarification 🙏